infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -127,6 +127,23 @@ boolean checkBootstrap(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r)
     // Helper Methods
 
 
+    @ServerHandler(type = CorfuMsgType.LAYOUT_FORCE, opTimer = metricsPrefix + "force")
+    public synchronized void handleLayoutForce(CorfuPayloadMsg<LayoutCommittedRequest> msg,
+                                               ChannelHandlerContext ctx, IServerRouter r,
+                                               boolean isMetricsEnabled) {
+        LayoutCommittedRequest req = msg.getPayload();
+
+        if (req.getEpoch() <= getServerEpoch()) {
+            // return can't force old epochs
+            r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.NACK));
+        }
+
+        setCurrentLayout(req.getLayout());
+        serverContext.setServerEpoch(req.getEpoch(), r);
```

Comment:
do I need to persist phase1/2 data too ?

Reply:
i don't think so but I do not remember to say with surity.